internal/discover/compat_libs.go

```diff
@@ -9,16 +9,12 @@ import (
 
 // NewCUDACompatHookDiscoverer creates a discoverer for a enable-cuda-compat hook.
 // This hook is responsible for setting up CUDA compatibility in the container and depends on the host driver version.
-func NewCUDACompatHookDiscoverer(logger logger.Interface, nvidiaCDIHookPath string, driver *root.Driver) Discover {
+func NewCUDACompatHookDiscoverer(logger logger.Interface, hookCreator HookCreator, driver *root.Driver) Discover {
 	_, cudaVersionPattern := getCUDALibRootAndVersionPattern(logger, driver)
 	var args []string
 	if !strings.Contains(cudaVersionPattern, "*") {
 		args = append(args, "--host-driver-version="+cudaVersionPattern)
 	}
 
-	return CreateNvidiaCDIHook(
-		nvidiaCDIHookPath,
-		"enable-cuda-compat",
-		args...,
-	)
+	return hookCreator.Create("enable-cuda-compat", args...)
```

Comment:
Note for a follow-up: `Create` should be modified to accept a `HookName` instead of arbitrary strings.

Reply:
I agree, but we would need to define the `HookName` in a new Package, as `pkg/nvcdi` calls `internal/discover` , so we can not reference `pkg/nvcdi` from `internal/discover` without running into an import loop. 
I think...

Reply:
I now have a better idea for this after our conversation, this will be addressed in a followup PR